vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/WalkableAggregate.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jackrabbit.vault.fs.impl.io;
```

Comment:
Is there a specific reason why we'd put an interface inside an `impl` package? It seems counterintuitive for me.

Reply:
yes. As I explained in the issue, the _API_ has a long history and I don't want to refactor it. I put this just besides the other ones, as it doesn't make sense to have them elsewhere.

if people want to use this library in a non-osgi environment, they can now implement their own `WalkableAggregates`, which wasn't possible before.